Kernel/Locking/LockRank.h

```diff
@@ -34,6 +34,9 @@ enum class LockRank : int {
     // Process locks are the highest rank, as they normally are taken
     // first thing when processing syscalls.
     Process = 0x010,
+
+    // Mutexes need to be taken before spinlocks, so they get their own lock-rank
```

Comment:
I'm having trouble following what this comment is trying to tell me? You shouldn't be allowed to acquire a mutex at all while you hold any spinlock, as that puts you in a position where you would be suspended while holding a spinlock.

Reply:
We don't have tracking for that IIRC, but if we check via requires() that all lock ranks of mutexes are at or above the spinlock ranks, that would give us a compile-time check for making sure that mutexes can even be taken legally.